addons/compositions/functions/fnc_openDisplay.sqf

```diff
@@ -111,21 +108,25 @@ private _ctrlButtonOK = _display displayCtrl IDC_OK;
     private _ctrlName     = _display displayCtrl IDC_DISPLAY_NAME;
 
     // Set the new composition category and name
-    _composition set [0, ctrlText _ctrlCategory];
-    _composition set [1, ctrlText _ctrlName];
+    private _category = ctrlText _ctrlCategory;
+    private _name = ctrlText _ctrlName;
+    private _compositionData = _composition select 2;
 
     if (_mode == "create") then {
         // In create mode, add the composition to saved data
         private _compositions = GET_COMPOSITIONS;
-        _compositions pushBack _composition;
-        SET_COMPOSITIONS(_compositions);
+        private _categoryHash = _compositions getOrDefault [_category, createHashMap, true];
+
+        _categoryHash set [_name, _compositionData];
+
+        SET_COMPOSITIONS(VAR_COMPOSITIONS);
     } else {
         // In edit mode, remove the old composition from the tree
         [false] call FUNC(removeFromTree);
     };
```

Comment:
Does this work in edit mode? I do not see where the existing composition's category and name are updated.

Reply:
In my testing, it seemed to work. I may have missed it though, was some time ago.

Reply:
The new code I pushed should solve that problem.

Reply:
Not fixed. Edit a composition by changing its name and category and reload the interface. Notice that the composition now exists under both the old and new name and category combinations.